### ISSUE TITLE
GPU card name to access 100X faster performance

### ISSUE THREAD

**Issue by Chitvan-baish** [deezer/spleeter#695]
I am using AWS instance g4dn.xlarge with Tesla T4 card, As I am unable to achieve 100x faster performance,
Anyone has suggestion which type of gpu card should I use to achieve 100x faster performance as per mention in the github library (https://github.com/deezer/spleeter).

Anyone has benchmarking on any gpu card. If yes, please share, It will help out. 

Thanks in advance

**Comment by Sameerhesta** [comment]
I am facing the same issue, I am splitting 1 min audio (Wav file) it is taking almost 12 to 15 seconds either I am using CPU or GPU.
It is not splitting my file as faster as it is coated in the documentation:-

> 2 stems and 4 stems models have high performances on the musdb dataset. Spleeter is also very fast as it can perform separation of audio files to 4 stems 100x faster than real-time when run on a GPU.

Can anyone guide me to achieve a faster separation speed?

**Comment by romi1502** [comment]
Hi @Chitvan-baish 
I addressed this problem in issue #697.
You should be able to do 100x faster than realtime if you don't count model loading overhead (which should be done only once for whatever the number of separation you want to perform).

**Comment by Chitvan-baish** [comment]
Hi @romi1502 , 
I know what are you trying to explain but I created pretrained model only once and after I run spleeter command 
first time it took 21 sec and then it took 13 sec on using gpu server of 1 min audio file

On cpu , it took 28 sec and then 14 sec (1min audio file)

**Comment by DevOps-Hestabit-1** [comment]
Hi @romi1502 
When i am running spleeter latest version 

here is my debug result, when running spleeter library

INFO:tensorflow:Using config: {'_model_dir': 'pretrained_models/4stems', '_tf_random_seed': None, '_save_summary_steps': 100, '_save_checkpoints_steps': None, '_save_checkpoints_secs': 600, '_session_config': gpu_options {
  per_process_gpu_memory_fraction: 0.7
}
, '_keep_checkpoint_max': 5, '_keep_checkpoint_every_n_hours': 10000, '_log_step_count_steps': 100, '_train_distribute': None, '_device_fn': None, '_protocol': None, '_eval_distribute': None, '_experimental_distribute': None, '_experimental_max_worker_delay_secs': None, '_session_creation_timeout_secs': 7200, '_checkpoint_save_graph_def': True, '_service': None, '_cluster_spec': ClusterSpec({}), '_task_type': 'worker', '_task_id': 0, '_global_id_in_cluster': 0, '_master': '', '_evaluation_master': '', '_is_chief': True, '_num_ps_replicas': 0, '_num_worker_replicas': 1}
WARNING:tensorflow:From /var/www/env1/lib/python3.7/site-packages/spleeter/separator.py:149: calling DatasetV2.from_generator (from tensorflow.python.data.ops.dataset_ops) with output_types is deprecated and will be removed in a future version.
Instructions for updating:
Use output_signature instead
WARNING:tensorflow:From /var/www/env1/lib/python3.7/site-packages/spleeter/separator.py:149: calling DatasetV2.from_generator (from tensorflow.python.data.ops.dataset_ops) with output_shapes is deprecated and will be removed in a future version.
Instructions for updating:
Use output_signature instead
INFO:tensorflow:Calling model_fn.
INFO:tensorflow:Apply unet for vocals_spectrogram
WARNING:tensorflow:From /var/www/env1/lib/python3.7/site-packages/tensorflow/python/keras/layers/normalization.py:534: _colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
INFO:tensorflow:Apply unet for drums_spectrogram
INFO:tensorflow:Apply unet for bass_spectrogram
INFO:tensorflow:Apply unet for other_spectrogram
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Restoring parameters from pretrained_models/4stems/model
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
DEBUG:spleeter:Writing file output5/temp11/vocals.wav
DEBUG:spleeter:Writing file output5/temp11/drums.wav
DEBUG:spleeter:Writing file output5/temp11/bass.wav
DEBUG:spleeter:Writing file output5/temp11/other.wav
INFO:spleeter:File output5/temp11/vocals.wav written succesfully
INFO:spleeter:File output5/temp11/drums.wav written succesfully
INFO:spleeter:File output5/temp11/bass.wav written succesfully
INFO:spleeter:File output5/temp11/other.wav written succesfully
Time taken to convert reduce the noise of audio file in seconds 13



is that, how is suppose to work or am I missing something because when i am seeing my gpu utilization goes upto 77%

**Comment by DevOps-Hestabit-1** [comment]
> Hi @Chitvan-baish I addressed this problem in issue #697. You should be able to do 100x faster than realtime if you don't count model loading overhead (which should be done only once for whatever the number of separation you want to perform).



Hi @romi1502
When i am running spleeter latest version

here is my debug result, when running spleeter library

INFO:tensorflow:Using config: {'_model_dir': 'pretrained_models/4stems', '_tf_random_seed': None, '_save_summary_steps': 100, '_save_checkpoints_steps': None, '_save_checkpoints_secs': 600, '_session_config': gpu_options {
per_process_gpu_memory_fraction: 0.7
}
, '_keep_checkpoint_max': 5, '_keep_checkpoint_every_n_hours': 10000, '_log_step_count_steps': 100, '_train_distribute': None, '_device_fn': None, '_protocol': None, '_eval_distribute': None, '_experimental_distribute': None, '_experimental_max_worker_delay_secs': None, '_session_creation_timeout_secs': 7200, '_checkpoint_save_graph_def': True, '_service': None, '_cluster_spec': ClusterSpec({}), '_task_type': 'worker', '_task_id': 0, '_global_id_in_cluster': 0, '_master': '', '_evaluation_master': '', '_is_chief': True, '_num_ps_replicas': 0, '_num_worker_replicas': 1}
WARNING:tensorflow:From /var/www/env1/lib/python3.7/site-packages/spleeter/separator.py:149: calling DatasetV2.from_generator (from tensorflow.python.data.ops.dataset_ops) with output_types is deprecated and will be removed in a future version.
Instructions for updating:
Use output_signature instead
WARNING:tensorflow:From /var/www/env1/lib/python3.7/site-packages/spleeter/separator.py:149: calling DatasetV2.from_generator (from tensorflow.python.data.ops.dataset_ops) with output_shapes is deprecated and will be removed in a future version.
Instructions for updating:
Use output_signature instead
INFO:tensorflow:Calling model_fn.
INFO:tensorflow:Apply unet for vocals_spectrogram
WARNING:tensorflow:From /var/www/env1/lib/python3.7/site-packages/tensorflow/python/keras/layers/normalization.py:534: _colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
INFO:tensorflow:Apply unet for drums_spectrogram
INFO:tensorflow:Apply unet for bass_spectrogram
INFO:tensorflow:Apply unet for other_spectrogram
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Restoring parameters from pretrained_models/4stems/model
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
DEBUG:spleeter:Writing file output5/temp11/vocals.wav
DEBUG:spleeter:Writing file output5/temp11/drums.wav
DEBUG:spleeter:Writing file output5/temp11/bass.wav
DEBUG:spleeter:Writing file output5/temp11/other.wav
INFO:spleeter:File output5/temp11/vocals.wav written succesfully
INFO:spleeter:File output5/temp11/drums.wav written succesfully
INFO:spleeter:File output5/temp11/bass.wav written succesfully
INFO:spleeter:File output5/temp11/other.wav written succesfully
Time taken to convert reduce the noise of audio file in seconds 13

is that, how is suppose to work or am I missing something because when i am seeing my gpu utilization goes upto 77%

**Comment by DevOps-Hestabit-1** [comment]
reference #418, In this, you mentioned 5 min audio will separate in ~3 sec, But In my case, it is taking 13 sec, Let me know how can I achieve this faster. (~3 sec)
